src/hotspot/share/classfile/klassFactory.cpp

```diff
@@ -109,27 +109,25 @@ InstanceKlass* KlassFactory::check_shared_class_file_load_hook(
 }
 
 
-static ClassFileStream* check_class_file_load_hook(ClassFileStream* stream,
+static ClassFileStream* check_class_file_load_hook(JavaThread* current,
+                                                   ClassFileStream* stream,
                                                    Symbol* name,
                                                    ClassLoaderData* loader_data,
                                                    Handle protection_domain,
-                                                   JvmtiCachedClassFileData** cached_class_file,
-                                                   TRAPS) {
```

Comment:
I don't think this removal of TRAPS is correct. The  ClassFileLoadHook could result in an exception becoming pending.

Reply:
Thanks!  This change was reverted.